packages/react-spinbutton/Spec.md

```diff
@@ -2,62 +2,366 @@
 
 - _Migration from v8_
+
+1. Ensure `value` prop is a number, not a string
+2. Replace `onIncrement` and `onDecrement` callbacks with `onChange`.
+   1. Increment/decrement logic can be handled by comparing `data.value` and `data.prevValue` in the `onChange` callback.
+3. Update `onChange` callback to handle new signature.
+4. Remove `onValidate` callback.
+5. Change ARIA props.
+   1. Change non-standard props to standard ones (e.g., `ariaLabel` to `aria-label`).
+   2. Remove `ariaPositionInSet`, `ariaSetSize`, `ariaValueNow`, `ariaValueText`. The first two are not relevant for a spinbutton and the latter two are internal implementation details managed by the component.
+
 - _Migration from v0_
 
+Not applicable as v0 does not implement this component or one like it.
+
 ## Behaviors
 
-_Explain how the component will behave in use, including:_
+`SpinButton`'s `value` prop is always a number, in contrast to the v8 implementation that gave `value` a string type. `SpinButton`s manipulate numeric values and making `value` a number aligns it with the other related props: `min`, `max` and `step`. `SpinButton`'s `value` is always displayed as a string which is determined with the `formatter()` function. By default this function stringifies `value` but it can do more sophisticated formatting like apply currency formatting. The inverse of `formatter()` is `parser()` which takes the stringified display value and converts it back to a number. The default implementation calls `parseFloat()` on the display value. `parser()` is called when a user directly types a value into `SpinButton`'s `<input>` element so this value can be converted to a number for computation. If `parser()` returns a non-number value (e.g., `NaN`, `undefined`, `null`) `value` will not be changed.
+
+`formatter()` and `parser()` allow users to hook into `SpinButton`'s display behavior without having to reimplement features like clamping `value` between `min` and `max` which is required when adding [custom suffixes in v8](https://codepen.io/seanms/pen/QWqpQWp).
+
+`SpinButton` does not allow values less than `min` or greater than `max` when these props are specified. When using the step buttons/arrow keys the value will not step outside the min/max bounds. If a value is typed into `<input>` that is outside the min/max range it will be clamped to the relevant bound, that is values less than `min` will be set to `min` and values greater than `max` will be set to `max`.
+
+Aside from min/max range clamping `SpinButton` does not currently implement any input validation.
+
+No error states are currently implemented.
+
+`SpinButton`'s `onChange` callback is invoked every time a change is committed. A change is committed when:
+
+1. A step button is activated
+2. When a user presses the `Arrow Up`, `Arrow Down`, `Home` or `End` keys while focused on the component.
+3. When a user presses the `Enter` key while focused on the `<input>`
```

Comment:
```suggestion
```

Let's remove `Enter` behaviour here, since I don't see it on native inputs: https://jsfiddle.net/wnydu8tc/